pkg/util/strategicpatch/patch.go

```diff
@@ -167,6 +213,12 @@ func mergeSlice(original, patch []interface{}, elemType reflect.Type, mergeKey s
 	if t.Kind() != reflect.Map {
 		// Maybe in the future add a "concat" mode that doesn't
 		// uniqify.
+		if !reflect.DeepEqual(original, patch) {
+			dw.WriteRemoval(fmt.Sprintf("%v", original))
+			dw.WriteAddition(fmt.Sprintf("%v", patch))
+		} else {
+			dw.WriteSame(fmt.Sprintf("%v", original))
+		}
```

Comment:
@caesarxuchao Then I need to take a step back to understand this. My understanding is that we're trying to produce a diff between `original` and `patch`. That's why in my previous comment:
* `a` has neither + nor - because it was in `original` and is still in `patch`
* `b` and `c` have a - because they were in `original` but not in `patch`, thus were removed
* `d` has a + because it was not in `original` but was `patch`, thus was added

Is my understanding wrong somewhere?